pkg/storage/allocator_scorer_test.go

```diff
@@ -757,6 +903,64 @@ func TestRemovalDiversityScore(t *testing.T) {
 	}
 }
 
+func TestDiversityScoreEquivalence(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	testCases := []struct {
+		stores   []roachpb.StoreID
+		expected float64
+	}{
+		{[]roachpb.StoreID{}, 1.0},
+		{[]roachpb.StoreID{testStoreUSa15}, 1.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe}, 0.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa1}, 0.25},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSb}, 0.5},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreEurope}, 1.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSa1}, 1.0 / 6.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSb}, 1.0 / 3.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreEurope}, 2.0 / 3.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa1, testStoreUSb}, 5.0 / 12.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa1, testStoreEurope}, 3.0 / 4.0},
+		{[]roachpb.StoreID{testStoreUSa1, testStoreUSb, testStoreEurope}, 5.0 / 6.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSa1, testStoreUSb}, 1.0 / 3.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSa1, testStoreEurope}, 7.0 / 12.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSb, testStoreEurope}, 2.0 / 3.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa1, testStoreUSb, testStoreEurope}, 17.0 / 24.0},
+		{[]roachpb.StoreID{testStoreUSa15, testStoreUSa15Dupe, testStoreUSa1, testStoreUSb, testStoreEurope}, 3.0 / 5.0},
+	}
+
+	// Ensure that rangeDiversityScore and diversityRebalanceFromScore return
+	// the same results for the same configurations, enabling their results
+	// to be directly compared with each other. The same is not true for
+	// diversityAllocateScore and diversityRemovalScore as of their initial
+	// creation or else we would test them here as well.
```

Comment:
I could understand if the remove target and the rebalance target are same, `rangeDiversityScore` will be same with `diversityRebalanceFromScore`, but I think maybe we should have a test that we could achieve better diversity score for`diversityRebalanceFromScore` if we remove one replica and rebalance to one target comparing with `rangeDiversityScore`.

Reply:
I don't think I understand exactly what you're suggesting. It sounds to me like you're suggesting something along the lines of what I've already added in `TestShouldRebalanceDiversity`, which is testing that new code in `shouldRebalance` is able to use `rangeDiversityScore` and `diversityRebalanceFromScore` to properly determine when a rebalance would be beneficial for diversity. Could you explain in more detail what you're suggesting and what code it would be testing?

Reply:
Hmm, Sorry for my bad expression. What I mean is that maybe we could add a test. The test will display the result of both functions, and literally the result of `diversityRebalanceFromScore` is obviously bigger than that of `rangeDiversityScore`.
`TestShouldRebalanceDiversity` proves that the result of which function is bigger but doesn't display the result of both functions.
Maybe it's unnecessary, I just think it's a better way for other guys to understand.

Reply:
Ok, I see now. That would help with understanding, but I don't think it provides any more code coverage. I'm going to skip it for now, but it would be a good test to have.